client/src/main/scala/drt/client/components/FlightChartComponent.scala

```diff
@@ -59,47 +70,65 @@ object FlightChartComponent {
                       val widthFactor = if (info.nationalities.size > 10 && state.showAllNationalities) 5 else 0
                       val chartWidth: Int = if (dom.window.innerWidth > 800)
                         300 + widthFactor * info.nationalities.size
-                        else
+                      else
                         200 + widthFactor * info.nationalities.size
 
-
-                      val paxTypeData: ChartJsData = ChartJsData(sortedPaxTypes.map {
-                        case (pt, _) => PaxTypes.displayNameShort(pt)
-                      }, sortedPaxTypes.map(_._2.toDouble), "Live API")
+                      val paxTypeData: ChartJsData = ChartJsData(
+                        labels = sortedPaxTypes.map {
+                          case (pt, _) => PaxTypes.displayNameShort(pt)
+                        },
+                        data = sortedPaxTypes.map(_._2.toDouble),
+                        dataSetLabel = "Live API",
+                        `type` = "bar")
 
                       <.div(^.cls := "container arrivals__table__flight__chart-box",
                         <.div(^.cls := "row", ^.width := (chartWidth * 3).toString + "px",
                           if (sortedNats.toMap.values.sum > 0)
                             <.div(^.cls := "col-sm arrivals__table__flight__chart-box__chart nationality-chart",
-                              ChartJSComponent.Bar(
+                              ChartJSComponent(
                                 ChartJsProps(
                                   data = nationalityData,
-                                  chartWidth,
-                                  chartHeight,
-                                  options = ChartJsOptions.withSuggestedMax("Nationality breakdown", sortedNats.toMap.values.max + 5)
+                                  width = Option(chartWidth),
+                                  height = Option(chartHeight),
+                                  options = ChartJsOptions("Nationality breakdown").copy(
+                                    scales = js.Dictionary[js.Any](
+                                      "y" -> js.Dictionary(
+                                        "suggestedMax" -> (sortedNats.toMap.values.max + 5),
+                                      ),
+                                    ))
                                 )
                               ))
                           else
                             EmptyVdom,
                           if (sortedPaxTypes.toMap.values.sum > 0 && sortedAges.toMap.values.sum > 0)
                             <.div(^.cls := "col-sm arrivals__table__flight__chart-box__chart passenger-type-chart",
-                              ChartJSComponent.Bar(
+                              ChartJSComponent(
                                 ChartJsProps(
```

Comment:
ChartJSComponent is repeatable . May be move to separate function with parameters ?